### PR TITLE
Final touches

### PR DESCRIPTION
**Pull request by wacoo**
- Add migrations, models, and controllers for group, expense, group_expense
- Add views for group, expense, group_expense, splash
- Add model, controller, and integration tests
- Add project presentation
- Fix linter errors

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -94,7 +94,15 @@ The app has the following features:
...
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
 
-<!-- GETTING STARTED -->
+## 🚀 Live Demo <a name="live-demo"></a>
+- Coming soon
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 💻 Project Presentation <a name="presentation"></a>
+- Click <a href="https://www.loom.com/share/1706468c115d406cb2b925df89fc3ba7?sid=26240e7e-77f7-4abe-ada3-39febc516083)">here</a> to see the presentation.
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
```
Comment (Qoosim) [inline README.md]: - [ ] Great job making a video highlighting the features of your application. However, there's a requirement that is missing from the video. It is required that your video should include highlight about a piece of the code you find interesting or you're proud of making. That is not yet included in your video. Kindly make a new video that includes that requirement.

piece-of-code

File: spec/features/expense/expense_integration_spec.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      expect(page).to have_content('Group ids')
+    end
+
+    scenario 'should show submit button' do
+      visit new_expense_path
+      sign_in(user1)
+      expect(page).to have_content('NEW EXPENSE')
+    end
+  end
+end
```
Comment (Qoosim) [inline spec/features/expense/expense_integration_spec.rb]: - [ ]  Some tests of your application are failing at this moment. Kindly look at them and fix. 

failed-test-1


failed-test-2

**Comment by Qoosim** [comment]
Hi @wacoo, you should not have submitted your project when you know that you're not fully done with the project. You are making new commits every now and then which has made the review to be difficult for me to give you feedback. Please, stop commiting to this PR again untill I am done with the review. Thanks

**Comment by wacoo** [comment]
Sorry, I forgot I am working on the same branch. I will not change it again.

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @wacoo, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  Implemented some project requirements. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/expense/expense_integration_spec.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  let(:user1) do
+    User.create(name: 'John', email: 'xyz@gmail.com', password: '123456', role: 'Admin')
+  end
+
+  before do
+    @user2 = User.create(name: 'Lily', email: 'abc@gmail.com', password: '123456')
+    @group1 = Group.create(name: 'Morgage', user: user1)
+    @group2 = Group.new(name: 'Transport', user: user1)
+    @group1.icon.attach(
+      io: File.open(Rails.root.join('app', 'assets', 'images', 'mo.png')),
```
Comment (Munsa1) [inline spec/features/expense/expense_integration_spec.rb:24]: - [ ] Good job for making the required changes however, we still have some failing testcases, Kindly resolve these issues as tests ensure our code works as expected.

image

Context
It looks like you are encountering an error related to the version of ChromeDriver in your Selenium configuration for testing. The error indicates that the specified version, 119.0.6045, is not found on the ChromeDriver storage.

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @wacoo, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: spec/features/expense/expense_integration_spec.rb (hunk 3 above)
Comment (wacoo) [reply spec/features/expense/expense_integration_spec.rb:24]: My laptop is Chromebook. Selenium doesn't detect the existing Chrome browser on my laptop or any Chrome browser that I install. The tests fails trying to load a browser. That's why I use rack_test which part of the same gem as selenium, but it doesn't load a browser. The tests are the same. However, you need to test each test type individually for it to work. Here are the same tests as the above: 
image

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Wonde,  

**RoR capstone project** is complete! 🏁 Great job, Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Test cases are passing!
🎲 ReadMe looks great!

Keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**